src/js/component.js

```diff
@@ -1247,21 +1245,18 @@ class Component {
   setTimeout(fn, timeout) {
     fn = Fn.bind(this, fn);
 
-    const timeoutId = window.setTimeout(fn, timeout);
```

Comment:
another option is to do something like:

```js
const timeoutId = window.setTimeout(() => {
  this.off('dispose', disposeFn);
  fn();
});
```

I think this would make sure that we don't keep the disposeFn around and we don't get into a situation where we accidentally don't call `fn` because of expectation changes for whether `this.player_` or `this.el_` exists.

Reply:
But disposeFn needs the timeoutId that's only available after setTimeout. I don't think you can get disposeFn into that closure after it was declared.

Reply:
maybe we need to declare them as `var`s rather than lets so things are available properly in scope with the value being set properly due to how those vars work.

Maybe there are other ways of doing it, not sure.